spec/requests/api/person_escort_records_controller_create_spec.rb

```diff
@@ -105,6 +105,47 @@
         it_behaves_like 'an endpoint that responds with error 404'
       end
 
+      context 'when a person escort record already exists on a profile' do
+        let(:errors_422) do
+          [
+            {
+              'title' => 'Unprocessable entity',
+              'detail' => 'Profile has already been taken',
+              'source' => { 'pointer' => '/data/attributes/profile' },
+              'code' => 'taken',
+            },
+          ]
+        end
+
+        before do
+          post '/api/v1/person_escort_records', params: person_escort_record_params, headers: headers, as: :json
+          post '/api/v1/person_escort_records', params: person_escort_record_params, headers: headers, as: :json
+        end
+
+        it_behaves_like 'an endpoint that responds with error 422'
+      end
+
+      context 'when a person escort record already exists on a profile and unique index error thrown' do
+        let(:errors_422) do
+          [
+            {
+              'title' => 'Invalid profile',
+              'detail' => 'Validation failed: Profile Profile has already been take',
+            },
+          ]
+        end
+
+        before do
+          person_escort_record = PersonEscortRecord.new
```

Comment:
attempted to use an `instance_double` but that didn't work as we will be calling methods regarding errors on this instance